src/pybind/mgr/dashboard_v2/models/nodb.py

```diff
@@ -0,0 +1,897 @@
+                    del meta_attrs[name]
+            #from django.db.models.options import DEFAULT_NAMES
+            #for attr_name in DEFAULT_NAMES:
+            #    if attr_name in meta_attrs:
+            #        setattr(self, attr_name, meta_attrs.pop(attr_name))
+            #    elif hasattr(self.meta, attr_name):
+            #        setattr(self, attr_name, getattr(self.meta, attr_name))
+
+            # unique_together can be either a tuple of tuples, or a single
+            # tuple of two strings. Normalize it to a tuple of tuples, so that
+            # calling code can uniformly expect that.
+            ut = meta_attrs.pop('unique_together', self.unique_together)
+            if ut and not isinstance(ut[0], (tuple, list)):
+                ut = (ut,)
+            self.unique_together = ut
+
+            # Any leftover attributes must be invalid.
+            if meta_attrs != {}:
+                raise TypeError("'class Meta' got invalid attribute(s): %s" % ','.join(meta_attrs.keys()))
+        del self.meta
+
+    @property
+    def concrete_fields(self):
+        return self.fields
+
+    @property
+    def concrete_model(self):
+        return self.model
+
+    # make django filter happy.
+    def get_field_by_name(self, name):
+        for field in self.fields:
+            if field.attname == name:
+                return field
+        raise KeyError('No field named "{}" in model "{}".'.format(name, str(self.model)))
+
+    def setup_pk(self, field):
+        if not self.pk and field.primary_key:
+            self.pk = field
+
+    def add_field(self, field):
+        self.fields.append(field)
+        self.setup_pk(field)
+
+
+
+class NodbModelMeta(type):
+    # Copy from django.db.models.base.ModelBase#__new__
+    def __new__(cls, name, bases, attrs):
+
+        # Create the class.
+        dunders = {k:v for k,v in attrs.items() if k in ['__module__', '__doc__', '__init__']}
+        for k in dunders:
+            del attrs[k]
+        new_class = super(NodbModelMeta, cls).__new__(cls, name, bases, dunders)
+        attr_meta = attrs.pop('Meta', None)
+        if not attr_meta:
+            meta = getattr(new_class, 'Meta', None)
+        else:
+            meta = attr_meta
+
+        # Figure out the app_label by looking one level up.
+        # For 'django.contrib.sites.models', this would be 'sites'.
+        new_class.add_to_class('_meta', NodbOptions(meta))
+        new_class.add_to_class('DoesNotExist', subclass('DoesNotExist', [NodbDoesNotExist]))
+        new_class.add_to_class('MultipleObjectsReturned', subclass('MultipleObjectsReturned', [NodbMultipleObjectsReturned]))
+
+        # Add all attributes to the class.
+        for obj_name, obj in attrs.items():
+            #if not obj_name.startswith('__'):
+            new_class.add_to_class(obj_name, obj)
+
+        new_class._meta._prepare(new_class)
+
+        if 'objects' not in attrs:
+            bases[0].objects.copy_manager_to(new_class)
+
```

Comment:
You will know what i mean :-)

Reply:
I took care of all linting errors today.